test/evergreen/print_python_stack_trace.py

```diff
@@ -0,0 +1,132 @@
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse
+import itertools
+import os
+import re
+import subprocess
+import sys
+from shutil import which
+
+
+def border_msg(msg: str):
+    count = len(msg) + 2
+    dash = "-"*count
+    return "+{dash}+\n| {msg} |\n+{dash}+".format(dash=dash, msg=msg)
+
+
+class LLDBDumper:
+    """LLDBDumper class."""
+    def __init__(self):
+        self.dbg = self.__find_debugger("lldb")
+
+    @staticmethod
+    def __find_debugger(debugger: str):
+        """Find the installed debugger."""
+        return which(debugger)
+
+    def dump(self, exe_path: str, core_path: str):
+        """Dump stack trace."""
+        if self.dbg is None:
+            sys.exit("Debugger gdb not found,"
+                     "skipping dumping of {}".format(core_path))
+
+        cmds = [
+            "thread backtrace all -c 30",
+            "quit"
+        ]
+
+        subprocess.run([self.dbg, "--batch"] + [exe_path, "-c", core_path] +
+                       list(itertools.chain.from_iterable([['-o', b] for b in cmds])),
+                       check=True)
+
+
+class GDBDumper:
+    """GDBDumper class."""
+    def __init__(self):
+        self.dbg = self.__find_debugger("gdb")
+
+    @staticmethod
+    def __find_debugger(debugger: str):
+        """Find the installed debugger."""
+        return which(debugger)
+
+    def dump(self, exe_path: str, core_path: str, lib_path: str):
+        """Dump stack trace."""
+        if self.dbg is None:
+            sys.exit("Debugger gdb not found,"
+                     "skipping dumping of {}".format(core_path))
+
+        cmds = []
+        if lib_path:
+            cmds.append("set solib-search-path " + lib_path)
+        cmds.extend([
+            "thread apply all backtrace 30",
+            "quit"
+        ])
+
+        subprocess.run([self.dbg, "--batch", "--quiet"] +
+                       list(itertools.chain.from_iterable([['-ex', b] for b in cmds])) +
+                       [exe_path, core_path],
+                       check=True)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-e', '--executable_path',
+                        help='path to the executable',
+                        required=True)
+    parser.add_argument('-c', '--core_path',
+                        help='directory path to the core dumps',
+                        required=True)
+    parser.add_argument('-l', '--lib_path', help='library path')
+    args = parser.parse_args()
+
+    # Store the path of the core files as a list.
+    core_files = []
+    regex = re.compile(r'.*dump.*python.*', re.IGNORECASE)
+    for root, _, files in os.walk(args.core_path):
```

Comment:
I'm curious why the 2nd tuple (subdirs) is ignored here.

Reply:
As the `subdirs` tuple is not required. Once the file name matches the regex, we simply join the `root` with the file name.

Reply:
Never mind. I was thinking of the possibility of the regex matched to a subdir, but likely not ('.*dump.*python.*' will be mapped to a file nearly certainly).